### PR TITLE
remove unnecessary elseif

### PR DESCRIPTION
**Pull request by thenotsoft**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

File: src/Rule/Email.php [hunk 1]
```diff
@@ -56,9 +56,11 @@ public function validateValue($value): Result
     {
         $result = new Result();
 
-        if (!is_string($value)) {
-            $valid = false;
-        } elseif (!preg_match('/^(?P<name>(?:"?([^"]*)"?\s)?)(?:\s+)?(?:(?P<open><?)((?P<local>.+)@(?P<domain>[^>]+))(?P<close>>?))$/i', $value, $matches)) {
+        if (!is_string($value) || !preg_match(
```
[the remark this answers is not in the document]
Comment (samdark) [reply src/Rule/Email.php]: Let's have both. Early return and combining these two checks.

**Comment by rob006** [comment]
Honestly, it is IMO **less** readable after these changes...

File: src/Rule/Email.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (kamarton) [reply src/Rule/Email.php]: > Let's have both. Early return and combining these two checks.

long line less readable.

I suggests:

```php
$errorResult = new ...->addError..
if(!...) {
  return $errorResult;
}
if(preg_match(self::RX_EMAIL, ...) ===0) {
  return $errorResult;
}
//... 
//... if success (eg. at end)
return new Result();
```

**Comment by samdark** [comment]
Yeah. I agree. Thanks for contribution, @thenotsoft but it won't be merged this time.

**Pull request closed without merging**